packages/ddp-server/livedata_server.js

```diff
@@ -330,9 +330,15 @@ var Session = function (server, version, socket, options) {
   self.send({ msg: 'connected', session: self.id });
 
   // On initial connect, spin up all the universal publishers.
-  Fiber(function () {
-    self.startUniversalSubs();
-  }).run();
+  if (Meteor._isFibersEnabled) {
+    Fiber(function() {
+      self.startUniversalSubs();
+    }).run();
+  } else {
+    ASL.run(Meteor._getAslStore, function () {
+      self.startUniversalSubs();
+    });
+  }
```

Comment:
A lot of code follows the same pattern, so it may be useful to create a function Meteor._runAsync which recognizes whether it should use Fibers or native Async/Await.

Maybe even better: we could create a wrapper package for the Fibers library (if not already there) and perform the check in there (so Fiber().run would use either Fibers or Async/Await)?

Reply:
@KoenLav Thanks for the suggestions! Yeah, I will try to create the wrapper and if it doesn't work, create a helper like you said. I'm just doing one by one for now to better understand the places that will be affected, since sometimes this requires the whole chain to be asynchronous.